### DISCUSSION TITLE
Shelly Plus 1PM: Show internal temperature

### DISCUSSION THREAD

**Discussion by GerdRuetten** [arendst/Tasmota#19203] [Q&A]
Hi all,

i flash a Shelly Plus 1PM with Tasmota 13.0.0

The shelly device has a internal temperature sensor (see: https://kb.shelly.cloud/knowledge-base/shelly-plus-1pm).

It is possible, to show the value from this sensor on the Tasmota main menu and publish this via mqtt?

Best regards

Gerd

**Comment by sfromis** [comment]
It is analog on gpio 32, if configured as `ADC Temp` and with the right [AdcParam](https://tasmota.github.io/docs/ADC/#commands) settings for temperature, to match the hardware. Haven't tried to dig up what the NTC settings should be.

On the non-PM [Shelly Plus 1](https://templates.blakadder.com/shelly_plus_1) "AdcParam1 2,10000,10000,3350" is suggested to get a "sensible value", but still needs calibration.

**Reply by GerdRuetten** [reply]
Dear @sfromis 

thank you very much for support!

I change under customizing -> template configuration the gpio32 to ADC Temp / 1 (see screenshot)

![image](https://github.com/arendst/Tasmota/assets/97361636/bd7c595b-9e01-4838-985e-41a4ee1cd336)

It show me the temperature, but it seems not correct (to low)

![image](https://github.com/arendst/Tasmota/assets/97361636/b5db2256-2936-43aa-afe0-3eacdf63fbe1)

The AdcParam1 are set to the default-values

`12:11:28.163 MQT: stat/tasmota_D7A388/RESULT = {"AdcParam1":[2,32000,10000,3350,0]}`

What have i do wrong?

Have you any idee?

Best regards

Gerd

**Reply by sfromis** [reply]
Did you try what I suggested, instead of the default values? With those, I'm seeing values like `"ANALOG":{"Temperature1":59.12}` from a Shelly Plus 1PM, but I've not tried to work out how "correct" this is.

**Reply by GerdRuetten** [reply]
> Did you try what I suggested, instead of the default values? With those, I'm seeing value ...

hello, I'm apparently blind 😂 

With this I get a temperatur near 48-50 C

**Reply by GerdRuetten** [reply]
For ESP32-Devices i found this Solution

https://tasmota.github.io/docs/ESP32/#cpu-temperature-sensor


Not nessssary to change the GPIO-Template

**Reply by sfromis** [reply]
Not really, that is such a bad "sensor" providing bogus values, that the "solution" is to keep it disabled.

**Reply by GerdRuetten** [reply]
> Not really, that is such a bad "sensor" providing bogus values, that the "solution" is to keep it disabled.

OK, hmmmm ... Thanks!
